src/AppBundle/Entity/EventRegistration.php

```diff
@@ -104,6 +105,15 @@ public function isNewsletterSubscriber(): bool
         return $this->newsletterSubscriber;
     }
 
+    public function isEventFinished(): bool
+    {
+        if (!$this->event instanceof Event) {
```

Comment:
Look at the JoinColumn constraint. In case of the event's deletion, the registration record is kept and the foreign key is set to null.